### ISSUE TITLE
Selecting cells using the table view has accessibility issues

### ISSUE THREAD

**Issue by simonwo** [register-dynamics/data-import#61]
![image](https://github.com/user-attachments/assets/4f034ed8-7169-465b-a7c6-e79eafd8bb17)

The table view is backed by a regular HTML table, and so is largely as accessible as any other table. However, selecting cells has some accessibility issues:

* Selecting cells can't be done with the keyboard (WCAG A)
* Dragging to select cells doesn't have a non-gesture alternative 
* It's not clear from display alone that cells are selectable 
* Cells don't have a maximum width and thus don't reflow (WCAG A)

(Aside: note that Reflow doesn't apply to the whole data table, just each cell:)
> Data tables have a two-dimensional relationship between the headings and data cells. This relationship is essential to convey the content. This Success Criterion therefore exempts data tables from needing to display without scrolling in the direction of text (e.g., horizontally in a vertically scrolling page). However, cells within data tables are not excepted unless the cell contains types of content that also requires two-dimensional layout for usage or meaning.

**Comment by simonwo** [comment]
Our existing range selection using the keyboard may actually be fine. We need to learn more about how aria-selected helps or does not, and how signal to the user agent that a cell is selectable if you click it. Probably we do have to introduce some input elements for selecting cells, or mark up cells with appropriate aria-roles (eg a checkbox role?)

Some discussion on appropriate aria-roles: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/grid_role
General tutorials on marking up accessible tables: https://www.w3.org/WAI/tutorials/tables/

**Comment by rossjones** [comment]
Some more useful links:

W3's web accessibility initiative [Grid (Interactive Tabular Data and Layout Containers) Pattern ](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)

Government Statistical Service (GSS) [Accessible spreadsheets : a checklist of the basics](https://gss.civilservice.gov.uk/policy-store/making-spreadsheets-accessible-a-brief-checklist-of-the-basics/)
